src/frontend/components/isbn-registry/subComponents/modals/SavePublisherModal/ModalComponent.jsx

```diff
@@ -95,7 +95,8 @@ function ModalComponent(props) {
       return <Spinner />;
     }
 
-    if (Object.keys(data).length === 0) {
+    // Display a message if object is empty, or data is not a type of object
+    if (Object.keys(data).length === 0 || typeof data !== 'object') {
```

Comment:
Check needs to be reversed: type test comes before testing number of object keys.